src/main/java/com/celements/payment/service/ComputopService.java

```diff
@@ -92,8 +91,126 @@ String hashPaymentData(@NotNull String paymentData) {
+    if (decryptedData != null) {
+      for (String parameter : (new String(decryptedData)).split("&")) {
+        String[] keyValue = parameter.split("=", 2);
+        String key = keyValue[0].toLowerCase();
+        String value = "";
+        if (keyValue.length > 1) {
+          value = keyValue[1];
+        }
+        callbackData.put(key, value);
+      }
+    }
+    return callbackData;
+  }
+
+  String getPaymentDataPlainString(String transactionId, String orderDescription, BigDecimal amount,
+      String currency) {
+    checkNotNull(transactionId);
+    checkNotNull(amount);
+    currency = Optional.fromNullable(currency).or(DEFAULT_CURRENCY);
+    String merchantId = getMerchantId();
+    StringBuilder sb = new StringBuilder();
+    sb.append(FORM_INPUT_NAME_MERCHANT_ID).append("=").append(merchantId);
+    sb.append("&").append(FORM_INPUT_NAME_TRANS_ID).append("=").append(transactionId);
+    sb.append("&").append(FORM_INPUT_NAME_AMOUNT).append("=").append(getAmount(amount));
+    sb.append("&").append(FORM_INPUT_NAME_CURRENCY).append("=").append(currency);
+    sb.append("&").append(FORM_INPUT_NAME_DESCRIPTION).append("=").append(orderDescription);
+    sb.append("&").append(FORM_INPUT_NAME_HMAC);
+    sb.append("=").append(getPaymentDataHmac(null, transactionId, merchantId, amount, currency));
+    sb.append("&").append(ReturnUrl.SUCCESS.getParamName());
+    sb.append("=").append(getReturnUrl(ReturnUrl.SUCCESS));
+    sb.append("&").append(ReturnUrl.FAILURE.getParamName());
+    sb.append("=").append(getReturnUrl(ReturnUrl.FAILURE));
+    sb.append("&").append(ReturnUrl.CALLBACK.getParamName());
+    sb.append("=").append(getReturnUrl(ReturnUrl.CALLBACK));
+    return sb.toString();
+  }
+
+  String encryptString(byte[] plainText, final SecretKey key) {
+    LOGGER.debug("encrypting plain [{}]", plainText);
+    try {
+      String cipherText = BaseEncoding.base16().encode(getCipher(Cipher.ENCRYPT_MODE,
+          BLOWFISH_ECB_PADDED, key).doFinal(plainText));
+      LOGGER.debug("encrypted cipher [{}]", cipherText);
+      return cipherText;
+    } catch (IllegalBlockSizeException | BadPaddingException | NoSuchPaddingException excp) {
+      LOGGER.error("Problem while encrypting payment data", excp);
+    }
+    return null;
+  }
+
+  byte[] decryptString(String encryptedBase16, int plainTextLength, final SecretKey key) {
+    CharSequence cs = encryptedBase16.toUpperCase();
+    LOGGER.debug("decrypting cipher [{}]", cs);
+    byte[] decodedCipher = BaseEncoding.base16().decode(cs);
+    try {
+      Cipher cipher = getCipher(Cipher.DECRYPT_MODE, BLOWFISH_ECB_UNPADDED, key);
+      byte[] deciphered = Arrays.copyOfRange(cipher.doFinal(decodedCipher), 0, plainTextLength);
+      LOGGER.debug("decryped plain [{}]", deciphered);
+      return deciphered;
+    } catch (IllegalBlockSizeException | BadPaddingException | NoSuchPaddingException excp) {
+      LOGGER.error("Problem while decrypting Computop callback", excp);
```

Comment:
I would prefer `decryptCallbackData`/`decryptString` to fail with an exception e.g. `ComputopEncryptionException` if the decryption fails